sdk/src/main/java/com/hedera/hashgraph/sdk/Transaction.java

```diff
@@ -792,16 +819,62 @@ public T freezeWith(@Nullable Client client) {
         return (T) this;
     }
 
-    void buildTransactions(int untilIndex) {
-        for (var i = transactions.size(); i < untilIndex; ++i) {
-            transactions.add(com.hedera.hashgraph.sdk.proto.Transaction.newBuilder()
-                .setSignedTransactionBytes(
-                    signedTransactions.get(i)
-                        .setSigMap(signatures.get(i))
-                        .build()
-                        .toByteString()
-                ).build());
+    /**
+     * Will build the specific transaction at {@code index} and will fill with `null` for any empty indices before it
+     * @param index
+     */
+    void buildTransaction(int index) {
+        if (index < 0) {
+            for (var i = 0; i < signedTransactions.size(); ++i) {
+                signTransaction(i);
+
+                transactions.add(com.hedera.hashgraph.sdk.proto.Transaction.newBuilder()
+                    .setSignedTransactionBytes(
+                        signedTransactions.get(i)
+                            .setSigMap(signatures.get(i))
+                            .build()
+                            .toByteString()
+                    ).build());
+            }
+
+            return;
+        }
```

Comment:
It seems like this should be factored into its own method, something like `buildAllTransactions()`, which then iterates and calls `buildTransaction(i)`